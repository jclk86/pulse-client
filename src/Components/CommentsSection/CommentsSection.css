.comments_list {
  margin: 0 auto;
  padding-left: 0;
}
