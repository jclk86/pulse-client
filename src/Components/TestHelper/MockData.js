export const mockUser = {
  username: "Pop123",
  password: "Password123!",
  fullname: "Bob Smith",
  email: "BSmith@gmail.com"
};

export const mockArticle = {
  title: "test article",
  content: "test content",
  article_tag: "News",
  id: 1,
  date_created: "2019-10-14T21:49:20.570Z",
  author: {
    username: mockUser.username
  }
};

export const mockComment = {
  id: 1,
  content: "test comment content",
  date_created: "2019-10-14T21:49:20.570Z",
  article_id: 1,
  user: {
    username: mockArticle.author.username
  }
};
