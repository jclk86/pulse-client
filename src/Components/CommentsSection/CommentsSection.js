import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import Comment from "../Comment/Comment";
import PropTypes from "prop-types";
import "./CommentsSection.css";

class CommentsSection extends Component {
  renderComments = comments => {
    return comments.map(comment => {
      return (
        <Comment
          comment={comment}
          key={comment.id}
          lightsOff={this.props.lightsOff}
        ></Comment>
      );
    });
  };

  render() {
    const { comments } = this.props;

    return (
      <div className="container_comments">
        <ul className="comments_list">
          {comments.length ? (
            this.renderComments(comments)
          ) : (
            <label>Add a comment!</label>
          )}
        </ul>
      </div>
    );
  }
}

CommentsSection.propTypes = {
  comments: PropTypes.arrayOf(
    PropTypes.shape({
      comment: PropTypes.string
    })
  ),
  lightsOff: PropTypes.bool
};

export default withRouter(CommentsSection);
