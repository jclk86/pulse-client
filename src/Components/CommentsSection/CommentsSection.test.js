import React from "react";
import { BrowserRouter } from "react-router-dom";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import AuthApiService from "../../Services/auth-api-service";
import ArticleApiService from "../../Services/article-api-service";
import CommentApiService from "../../Services/comment-api-service";
import { mockUser, mockArticle, mockComment } from "../TestHelper/MockData";
import CommentsSection from "./CommentsSection";

describe(`CommentsSection`, () => {
  beforeAll(() => {
    AuthApiService.postUser(mockUser).then(user => {
      AuthApiService.postLogin({
        username: user.username,
        password: user.password
      });
    });

    ArticleApiService.postArticle(mockArticle).then(() => {
      CommentApiService.postComment(mockComment);
    });
  });

  it("renders CommentSection without issues", () => {
    const wrapper = shallow(
      <BrowserRouter>
        <CommentsSection comments={mockComment}></CommentsSection>
      </BrowserRouter>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
