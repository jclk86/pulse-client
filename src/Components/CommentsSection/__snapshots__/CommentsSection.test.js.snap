// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsSection renders CommentSection without issues 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <withRouter(CommentsSection)
    comments={
      Object {
        "article_id": 1,
        "content": "test comment content",
        "date_created": "2019-10-14T21:49:20.570Z",
        "id": 1,
        "user": Object {
          "username": "Pop123",
        },
      }
    }
  />
</Router>
`;
